pyafq_cli.json

```diff
@@ -0,0 +1,501 @@
+{
+    "name": "pyAFQ",
+    "tool-version": "0.1.1",
+    "description": "Automated Fiber Quantification ... in Python",
+    "command-line": "pyAFQ [CONFIG_FILE]",
+    "schema-version": "0.5",
+    "container-image": {
+        "type": "docker",
+        "image": "gkiar/fuzzy:ubuntu14.04-python-dipy-afq"
+    },
+    "groups":[
+    {
+        "id": "bundles_group",
+        "name": "Bundles Group",
+        "description": "Parameters related to bundles of interest.",
+        "members": ["bundles", "bundles_seg_algo", "scalars_model", "scalars"]
+    },
+    {
+        "id": "tracking_group",
+        "name": "Tracking Group",
+        "description": "Parameters related to streamline tracking.",
+        "members": ["directions", "max_angle", "sphere", "seed_mask", "n_seeds",
+                    "random_seeds", "stop_mask", "stop_threshold", "step_size",
+                    "min_length", "max_length", "odf_model", "wm_labels"]
+    },
+    {
+        "id": "segmentation_group",
+        "name": "Segmentation Group",
+        "description": "Parameters related to segmentation.",
+        "members": ["nb_points", "segmentation_seg_algo", "progressive",
+                    "greater_than", "rm_small_clusters", "model_clust_thr",
+                    "reduction_thr", "refine", "pruning_thr", "b0_threshold",
+                    "prob_threshold", "rng", "segmentation_return_idx",
+                    "filter_by_endpoints", "dist_to_aal"]
+    },
+    {
+        "id": "cleaning_group",
+        "name": "Cleaning Group",
+        "description": "Parameters related to cleaning.",
+        "members": ["n_points", "clean_rounds", "distance_threshold",
+                    "length_threshold", "min_sl", "stat", "cleaning_return_idx"]
+    }
+    ],
+    "inputs": [
+    {
+        "id": "config_file",
+        "name": "Configuration file",
+        "type": "String",
+        "value-key": "[CONFIG_FILE]",
+        "optional": false
+    },
+    {
+        "id": "dmriprep_path",
+        "name": "dmriprep path",
+        "type": "String",
+        "value-key": "[DMRIPREP_PATH]",
+        "optional": false
+    },
+    {
+        "id": "bundles",
+        "name": "bundles",
+        "type": "String",
+        "list": true,
+        "min-list-entries": 1,
+        "list-separator": "', '",
+        "value-key": "[BUNDLES]",
+        "value-choices": ["ATR", "CGC", "CST", "HCC", "IFO", "ILF",
```

Comment:
> IIUC, you are correct. The bundles are selected from a rather strictly defined set. Would it be helpful if the possible values of these strings were stored somewhere?
I currently have the set of all possible bundles set here, which is also the default value. If there are other bundles not in the default we can just add them to this list in `value-choices` but not `default-value`.

Reply:
This seems fine for now. 

It does get a bit complicated: First, we might have additional bundles in the future. Second, the Recobundles segmentation algorithm uses a different list as its default. How do you think we should handle this?

Reply:
Is the list overlapping with this one? Perhaps we could merge the two lists in the `value-choices` field, just to cover all the bases we currently have, and then leave the `default` as-is. Thoughts?

Reply:
There is some overlap. The full list for recobundles is: `['CST_R', 'C_L', 'F_L', 'UF_R', 'MCP', 'AF_R', 'CCMid', 'F_R', 'UF_L', 'AF_L', 'CC_ForcepsMajor', 'CC_ForcepsMinor', 'CST_L', 'IFOF_L', 'C_R', 'IFOF_R']`.  Note that even where there is overlap (e.g., `'CST'`), it is not clear that they stand for exactly the same thing. That's part of the work we are doing now: figuring out how these segmentation  algorithms differ in terms of their results, the reliability of these results and so on.

Reply:
BTW, this list is derived from: 

```
import AFQ.data as afd
reco_bundles = afd.read_hcp_atlas_16_bundles()      
reco_bundles.keys()
```

Reply:
Cool, thanks! For now, I've added this list to the possible value choices and we can re-evaluate if we find a change needs to be made